packages/docusaurus-utils/src/tags.ts

```diff
@@ -128,3 +130,32 @@ export function groupTaggedItems<Item>(
 
   return result;
 }
+
+/**
+ * Permits to get the "tag visibility" (hard to find a better name)
+ * IE, is this tag listed or unlisted
+ * And which items should be listed when this tag is browsed
+ */
+export function getTagVisibility<Item>({
+  items,
+  isUnlisted,
+}: {
+  items: Item[];
+  isUnlisted: (item: Item) => boolean;
+}): {
+  unlisted: boolean;
+  listedItems: Item[];
+} {
+  const allItemsUnlisted = items.every(isUnlisted);
+  // When a tag is full of unlisted items, we display all the items
+  // when tag is browsed, but we mark the tag as unlisted
+  if (allItemsUnlisted) {
+    return {unlisted: true, listedItems: items};
+  }
```

Comment:
This logic is questionable.

If 2 unlisted content (blog/docs) have the same tag, then the tag gets unlisted (ie it does not appear in the tags index, noindex...).

But there's a problem: now if you share URL of content1 to somehow, they are able to navigate the tag and see that content2 also exists. Is that a problem? Do we want to prevent unlisted content to be discovered that way?

Note we really have to render the unlisted tag page, otherwise we'd have a broken link (because both content have a link to that tag page)


---

Basically, there are 3 choices:
- an unlisted tags list all the unlisted content that has that tag (current implementation)
- we filter unlisted tags from content so that there's no need to create the tag page, and we don't have any broken link ➡️ may be confusing for users trying to add new tags to an unlisted document?
- we render a tag page, but display no content inside ➡️ user clicking on a content's unlisted tag will not see the content they came from, also a bit confusing. In this case it's also possible to display a better tag-related message (not the generic unlisted one) explaining what is happening

For now I'll keep the first implementation, that remains quite an edge case and maybe nobody will ever notice about this subtility...